Dockerfile

```diff
@@ -20,8 +20,24 @@ RUN DEBIAN_FRONTEND=noninteractive apt-get update -yqq && \
     python3-pylibacl \
     python3-pyxattr
 
+# Build dependencies specific for rdiff-backup development and testing
+RUN DEBIAN_FRONTEND=noninteractive apt-get update -yqq && \
+    apt-get install -y --no-install-recommends \
+    tox \
+    rdiff \
+    libacl1-dev # /usr/include/sys/acl.h
+ 
 # Build dev image
 # docker build --pull --tag rdiff-backup-dev:debian-sid .
 
-# Build rdiff-backup
-# docker run -it -v ${PWD}:/build -w /build rdiff-backup-dev:debian-sid ./setup.py build
+# Build rdiff-backup (assumes source is in directory 'rdiff-backup' and it's parent is writeable)
+# docker run -it -v ${PWD}/..:/build -w /build/rdiff-backup rdiff-backup-dev:debian-sid ./setup.py build
+
+# Run tests
+# docker run -it -v ${PWD}/..:/build -w /build/rdiff-backup rdiff-backup-dev:debian-sid bash
+#   cd ..
+#   curl -LO https://github.com/ericzolf/rdiff-backup/releases/download/Testfiles2019-08-10/rdiff-backup_testfiles_2019-08-10.tar.gz
+#   tar xvf *.tar.gz
+#   cd rdiff-backup
+#   tox -e py37
+
```

Comment:
`sudo tox -c tox_root.ini -e py37` is missing (it uncovers a few things).
Not sure about `tox -c tox_slow.ini` (it's slow).

Reply:
Tests now produce:
```
root@6471d534de6a:/build/rdiff-backup# tox -c tox_root.ini -e py37
GLOB sdist-make: /build/rdiff-backup/setup.py
py37 create: /build/rdiff-backup/.tox.root/py37
py37 installdeps: pyxattr, pylibacl
py37 inst: /build/rdiff-backup/.tox.root/.tmp/package/1/rdiff-backup-1.3.4.zip
py37 installed: pkg-resources==0.0.0,pylibacl==0.5.3,pyxattr==0.6.1,rdiff-backup==1.3.4
py37 run-test-pre: PYTHONHASHSEED='733931859'
py37 run-test-pre: commands[0] | python testing/commontest.py
py37 runtests: commands[0] | python testing/roottest.py
Traceback (most recent call last):
  File "testing/roottest.py", line 22, in <module>
    userid = int(os.getenv('RDIFF_TEST_UID', os.getenv('SUDO_UID')))
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
ERROR: InvocationError for command '/build/rdiff-backup/.tox.root/py37/bin/python testing/roottest.py' (exited with code 1)
```

Reply:
```root@6471d534de6a:/build/rdiff-backup# echo $RDIFF_TEST_UID

root@6471d534de6a:/build/rdiff-backup# echo $SUDO_UID
```

Reply:
Yes, if you don't use `sudo` you need to explicitly define the normal user's  ID with the RDIFF_TEST_UID environment variable.

Reply:
So the test requires that is must be run by a normal user? Any requirements on what kind of user that "normal" user is?

Is there something else one needs to know about how to run the tests or what are the test environment requirements?

The goal with this Dockerfile is to provide an environment anybody can reproduce, so all things need to be documented here now in detail about the environment and the commands to run.